scripts/set-version.js

```diff
@@ -0,0 +1,163 @@
+const path = require('path');
+const replace = require('replace-in-file');
+const moment = require('moment');
+const {
+  displayConfirmationMessage,
+  displayErrorMessage
+} = require('./common');
+const hotPackageJson = require('../package.json');
+
+const CURRENT_VERSION = hotPackageJson.version;
+const SEMVER_SECTIONS = ['major', 'minor', 'patch'];
+
+/**
+ * Check if the provided version number is a valid semver version number.
+ *
+ * @param {string} version Version number.
+ * @returns {boolean} `true` if the version number is a valid semver version number, `false` otherwise.
+ */
+function isVersionValid(version) {
+  const semverRegex = new RegExp(
```

Comment:
The [SemVer](https://www.npmjs.com/package/semver) package should do the same and more. It has only one dependency so shouldn't affect the install time much.